pygsti/layouts/matrixlayout.py

```diff
@@ -78,24 +82,29 @@ def add_expanded_circuits(indices, add_to_this_dict):
             for i in indices:
                 nospam_c = unique_nospam_circuits[i]
                 for unique_i in circuits_by_unique_nospam_circuits[nospam_c]:  # "unique" circuits: add SPAM to nospam_c
-                    observed_outcomes = None if (dataset is None) else dataset[ds_circuits[unique_i]].unique_outcomes
```

Comment:
Question for folks, but particularly aimed at @enielse. A comparable line to this one for computing the quantity named `observed_outcomes` appears [here](https://github.com/sandialabs/pyGSTi/blob/53e2da62422699e0af04dafd588776a7abeec7cc/pygsti/layouts/maplayout.py#L61), but in the map layout code this queries the `outcomes` property of `DataSetRow` instead of the `unique_outcomes` property as is done here. I think this should be `unique_outcomes` in both cases (I don't see a reason why it wouldn't also be for map), but I wanted to run this by you before I made that change.